README.md

```diff
@@ -1,22 +1,119 @@
 # iobroker-object-helper
+This helper supports Adapter developers for ioBroker with an easy interface for ioBroker objects. The main features are:
+* Create objects based on templates or roles defined in the official documentation with a single function call:
+    * https://github.com/ioBroker/ioBroker.docs/blob/master/docs/en/dev/objectsschema.md
+    * https://github.com/ioBroker/ioBroker.docs/blob/master/docs/en/dev/stateroles.md
+* Save a complete array of objects to ioBroker with a single function call
+    * Object tree gets validated, so all devices, channels or folders in the path must be defined
+    * No longer required objects in ioBroker could be removed automatically, exceptions as string or regex possible
+    * Overwrite ioBroker objects just as parameter
 
-## Example usage:
+Credits for JeyCee, who created the basic libraries we customized for this project:
+https://github.com/iobroker-community-adapters/iobroker-adapter-helpers/tree/master/definitions
 
-### Creates initial array with a channel named "user":
+## Use in Adapter project
 
-`const iobResult: iobObjectHelper.iobObject[] = Array(iobObjectHelper.makeIOBObj("user", "user", null, "template", "channel"));`
+Installation e.g. in Terminal window in VSCode:
+
+```
+npm install gaudes/iobroker-object-helper
+```
+
+An entry in package.json will be created:
+
+```
+  "dependencies": {
+    "@iobroker/adapter-core": "^2.4.0",
+    "iobroker-object-helper": "github:gaudes/iobroker-object-helper",
+    ...
+  },
+```
+
+## Integration
+
+For TypeScript-Adapter use
+```typescript
+import * as iobObjectHelper from "iobroker-object-helper";
+```
+For JavaScript-Adapter use
+```javascript
+const iobObjectHelper = require("iobroker-object-helper");
+```
+
+## Functions
+
+### buildObject
+This functions returns a validated ioBroker object.
+
+It has two main features:
+* Create object based on a template
+* Create a custom object based on a defined role
+
+Example usage:
+
+#### Create initial array with a channel named "user":
+
+TypeScript
+```typescript
+const iobResult: iobObjectHelper.ObjectWithValue[] = Array(iobObjectHelper.buildObject(this,{id: "user", name: "user", objectType: "template", template: "channel"}));
+```
+Javascript
+```javascript
+const iobResult = Array(iobObjectHelper.buildObject(this,{id: "user", name: "user", objectType: "template", template: "channel"}));
```

Comment:
and here?

Reply:
I tried to provide a complete exampel. Create an array of objects and Save the complete array

Reply:
Then I would use 2 objects so the purpose of the array is clear 😅

Reply:
Ah, First example is TS and second is JS. Ist should be an easy description also for beginners. So I tried to make examples with both languages

Reply:
This is okay, but I would still use arrays with at least two objects, so one doesn't have to wonder why the array is used at all.
Also it is more common to define arrays with `[ ...elements ]` instead of `Array(...elements)`.